packages/common/src/utils/SafeLocalStorage.ts

```diff
@@ -30,14 +30,29 @@ export const SafeLocalStorage = {
     value: SafeLocalStorageItems[Key]
   ): void {
     if (isSafe()) {
-      localStorage.setItem(key, value)
+      localStorage.setItem(
+        key,
+        key === SafeLocalStorageKeys.DEEPLINK_CHOICE ? JSON.stringify(value) : (value as string)
+      )
     }
   },
   getItem<Key extends keyof SafeLocalStorageItems>(
     key: Key
   ): SafeLocalStorageItems[Key] | undefined {
     if (isSafe()) {
-      return localStorage.getItem(key) || undefined
+      const value = localStorage.getItem(key)
+
+      if (value) {
+        if (key === SafeLocalStorageKeys.DEEPLINK_CHOICE) {
+          try {
+            return JSON.parse(value)
+          } catch {
+            return undefined
+          }
+        }
+
+        return value as SafeLocalStorageItems[Key]
+      }
```

Comment:
I wanted to use `JSON.parse` and `JSON.stringify` to not make the code complex, but if the devs were to upgrade their appkit dependency it'd cause issues with their current local storage. 

Instead i applied this logic to `SafeLocalStorageKeys.DEEPLINK_CHOICE` only and the rest of `SafeLocalStorageKeys` should return as is to make it backward compatible.